airflow/opt/providers/etna/etna/etls/box.py

```diff
@@ -0,0 +1,127 @@
+        folder path for each file for further filtering.
+
+        args:
+            files: List of files from tail_files or previous filter_files call
+            file_name_regex: re.Pattern, i.e. re.compile(r".*\.fcs")
+            folder_path_regex: re.Pattern, i.e. re.compile(r".*ipi.*")
+        """
+
+        @task
+        def filter_files(files):
+            result: List[FtpEntry] = [f for f in files if folder_path_regex.match(f.folder_path) and file_name_regex.match(f.name)]
+
+            return result
+
+        return filter_files(files)
+
+    def ingest_to_metis(
+        self,
+        files: XComArg,
+        project_name: str = "triage",
+        bucket_name: str = "waiting_room",
+        folder_path: str = None,
+        flatten: bool = True,
+        clean_up: bool = False) -> XComArg:
+        """
+        Given a list of Box files, will copy them to the given Metis project_name and bucket_name,
+        mimicking the full directory structure from Box.
+
+        args:
+            files: List of files from tail_files or filter_files call
+            project_name: str, the target Metis project name. Default is `triage`
+            bucket_name: str, the target Metis bucket name. Default is `waiting_room`
+            folder_path: str, existing folder path to dump the files in. Default is Box hostname + folder structure in Box.
+            flatten: bool, to flatten the Box folder structure or maintain it. Default is True.
+            clean_up: bool, to remove the file from Box after ingest. Default is False.
+        """
+        @task
+        def ingest(files, project_name, bucket_name, folder_path):
+            etna_hook = EtnaHook.for_project(project_name)
+            with etna_hook.metis(project_name, read_only=False) as metis, self.hook.box() as box, box.ftps() as ftps:
+                self.log.info(f"Attempting to upload {len(files)} files to Metis")
+                for file in files:
+                    with box.retrieve_file(ftps, file) as box_io_file:
+                        parts = []
+                        if folder_path is None:
+                            parts.append(self.hook.connection.host)
+                        else:
+                            parts.append(folder_path)
+
+                        if flatten:
+                            parts.append(file.name)
+                        else:
+                            parts.append(file.rel_path)
+
+                        dest_path = os.path.join(*parts)
+
+                        self.log.info(f"Uploading {file.full_path} to {dest_path}.")
+
+                        for blob in metis.upload_file(
+                            project_name,
+                            bucket_name,
+                            dest_path,
+                            box_io_file,
+                            file.size
+                        ):
+                            self.log.info("Uploading blob...")
```

Comment:
Kind of silly, but I wonder if we should limit the logging by time and not just blob iterations?  Like log once every 3-5 seconds?  Haha, just that the same message logging over and over for each chunk probably is overkill?

I realize the upload_file iterator interface is odd, but I found it to be most technically correct at the time.